### ISSUE TITLE
Zipping the archives is overriding the already zipped logs

### ISSUE THREAD

**Issue by gitmnikhil** [winstonjs/winston-daily-rotate-file#99]
We have a scenario where we are rotating the logs every 10Mb. When the number of request are high the log file is generated almost every second.

We have set zippedArchive to true. On first file it successfully zip the file and create a new file. But as soon as another file is ready to be archived, It overwrites the previous one as they have the same name instead of creating a new archive.

Config
```javascript
new winston.transports.DailyRotateFile({
      name: 'info-file',
      filename: 'logs/info.log',
      level: loggerConfig.level,
       datePattern: 'yyyy-MM-dd.',
       colorize: true,
       level: loggerConfig.level,
       prepend: true,
       // 10 MB max file size then rotate
       maxsize: 10485760
    })

**Comment by gitmnikhil** [comment]
@mattberther Can you please help, if this is a configuration issue or confirm if this is the module bug.

**Comment by nirmal-das** [comment]
We observed that the application exits with exception when max size is reached and zipping is being performed. 

Following exception was observed:
> Error: EPERM: operation not permitted, stat 'F:\Project Code\logs\app-logger\2017-11-03.info.log

Stack strace

> C:\Users\Nirmal.COMPRO\AppData\Roaming\npm\node_modules\pm2\lib\API\Extra.js:70
        fmt.field('uid', process.geteuid());
                                 ^
TypeError: process.geteuid is not a function
    at C:\Users\Nirmal.COMPRO\AppData\Roaming\npm\node_modules\pm2\lib\API\Extra.js:70:34
    at C:\Users\Nirmal.COMPRO\AppData\Roaming\npm\node_modules\pm2\node_modules\pm2-axon-rpc\lib\client.js:45:10
    at Parser.<anonymous> (C:\Users\Nirmal.COMPRO\AppData\Roaming\npm\node_modules\pm2\node_modules\pm2-axon\lib\sockets\req.js:67:8)
    at emitOne (events.js:115:13)
    at Parser.emit (events.js:210:7)
    at Parser._write (C:\Users\Nirmal.COMPRO\AppData\Roaming\npm\node_modules\pm2\node_modules\amp\lib\stream.js:91:16)
    at doWrite (_stream_writable.js:387:12)
    at writeOrBuffer (_stream_writable.js:373:5)
    at Parser.Writable.write (_stream_writable.js:290:11)
    at Socket.ondata (_stream_readable.js:642:20)

**Configuration used:**  
```
new winston.transports.DailyRotateFile({
        name: 'error-file',
        filename: path.resolve(loggerConfig.directory, "err.log"),
        handleExceptions: true,
        humanReadableUnhandledException: true,
        level: 'error',
        datePattern: 'yyyy-MM-dd.',
        prepend: true,
        maxsize: 307200,
        zippedArchive: true
    })
```

**Debugging Info**
_createStream is called based on comparison of maxsize with buffer size (this._size)
![image](https://user-images.githubusercontent.com/22272864/32370961-f60d8f54-c0b4-11e7-85cf-da2e8abf1fb4.png)

However, in comparison of maxsize is done with stat.size (which is smaller) and hence log file to be used next does not get incremented name 
![image](https://user-images.githubusercontent.com/22272864/32371024-43af11ec-c0b5-11e7-8856-7d3109efdfa5.png)

Finally, the exception 
![image](https://user-images.githubusercontent.com/22272864/32371035-54c3b334-c0b5-11e7-9778-66329383f529.png)

May be use buffer size (this._size) at line# 601 instead of stat.size?

**Comment by mattberther** [comment]
Some long-overdue refactoring of the transport has occurred. A PR is available at #107. Also, a pre-release version has been published to npm. Please `npm install winston-daily-rotate-file@2.0.0-beta` and reopen this issue if the problem remains.

**Comment by adwiens** [comment]
We had the same issue as @nirmaldas84 with EPERM when zipping. Upgrading from 1.4.6 to latest (3.2.1 at time of writing) solved it for us.
